### ISSUE TITLE
[quick question] What is the difference between mup and mupx?

### ISSUE THREAD

**Issue by y-ich** [arunoda/meteor-up-legacy#903]
I am confused to see two repositories, this one and https://github.com/kadirahq/meteor-up.

**Comment by MasterJames** [comment]
Well mup uses upstart while this was good and still prefered by some many problems were solved by mupx. Mupx is a docker centric approach. Docker containers take a little getting use to as they seem isolated and hard to access but once you know how it's possible the opposite is true. Using commands like "docker exec" you can open a bash shell or just execute commands like ls ps etc if they have been made available of course.
It made sense to rebrand it with kadirahq as it is a pretty complex suite of interconnected packages if you dig around. 
Anyway also mupx uses nginx.
In the near future meteor 1.3 will have ssl available through a newer version of nodejs. Definitely you'll want to know and use Docker or container logic weather it be microservices or what.
I've noted that once you are deployed you could just overwrite the files with rcp and restart. I even built my own file watch system like meteor does to rebuild the deployment from src but it just reloads files and overwrites the object structure they extend. So really once it's up you could make changes in other ways and in a sense you can avoid other problems. 
Mongo mounts a volume with Docker run command so those files are actually always available outside of the container and persist redeployment.
Anyway I hope that gives you a good base to learn more and maybe avoid a few more inquires. Maybe you'll be submitting PRs to the good people at kadirahq but unfortunately this version is depreciated by it and no longer merging fixes but it is at this time still the most production ready although I don't think any of them make that claim they are working pretty Darn well.

**Comment by pbreit** [comment]
I'm still confused. Is there a recommendation on which version to use today?

**Comment by MasterJames** [comment]
Apparently it's still this arunoda but the development branch mupx. I figure it's best to work with the new branch and the available team to sort out smaller issues you have with it. If it can't cut you can come back here. Alternatively of this one is working great but if you want/need changes or support its all dedicated to kadirahq version now.
BTW I'm no official voice just here to help and migration to the new version is my hope for everyone.

**Comment by pbreit** [comment]
Should I be able to use the same mup.json file "as-is" with mup and mupx? I noticed the sample file in the mupx readme is missing a few items like setupNode, etc.

**Comment by MasterJames** [comment]
Similar but best to manually adjust your settings from the new default one.
It's a js file now not a json file.

**Comment by pbreit** [comment]
Yes, that makes sense. But I was worried about not including things like setupNode and nodeVersion (which are missing from the mupx example). Are those no longer necessary?

**Comment by MasterJames** [comment]
Right that's true it's in the docker image used now.

**Comment by nickredmark** [comment]
Subscribing to this. Please update when the kadirahq becomes the standard :)

**Comment by MasterJames** [comment]
Mupx and kadirahq port are docker centric and nginx for ssl. Mup uses stud.
The new version has loads of important needed features especially moving forward.

**Comment by nickredmark** [comment]
The readme of kadirahq still says

"This version of Meteor Up is still in development. Please check arunoda/meteor-up for the stable version."

@MasterJames Have you been using the kadirahq version?

**Comment by MasterJames** [comment]
No. Kind of waiting for Meter 1.3.?  to stabilize and be well integrated there.I expected a TLS version of node integrated as well,  but its still in progress at meteor. Those comments are in regards to stability meaning will not change or is changing lots. 
These versions are depreciated by the new kadirahq version.
I'm encouraging people to go there and try it and work with them to address the various situations that arise so it's better sooner faster.it should already be better because of loads of features. Hooks and custom docker images are two that can solve loads of problems. 
To be honest I see it as a good starting point to learn and modify as needed. Being open source PRs are only being accepted in the new version that was built for 1.2 so I find myself trying to bring awareness to others about the new version because most of the issues posted here i fear will go unanswered otherwise.
